### PR TITLE
Merge master

### PR DESCRIPTION
**Pull request by jyotimahapatra**
@jessicayuen Since the xdsimpl interface changed quite a bit, im trying to help with the merge of master into your orchestrator branch.

### CODE REVIEW

File: internal/app/orchestrator/orchestrator.go [hunk 1]
```diff
@@ -187,24 +187,24 @@ func (o *orchestrator) Fetch(context.Context, discovery.DiscoveryRequest) (*gcp.
...
 	for {
 		select {
-		case x := <-responseChannel:
-			if x.Err != nil {
+		case x, more := <-responseChannel:
+			if !more {
 				// A problem occurred fetching the response upstream, log and
 				// return the most recent cached response, so that the
 				// downstream will reissue the discovery request.
-				o.logger.With("err", x.Err).With("key", aggregatedKey).Error(ctx, "upstream error")
```
Comment (jessicayuen) [inline internal/app/orchestrator/orchestrator.go:200]: Did we get rid of the `With("err", x.Err)` because it's a duplicate? Should still log the key imo.

**Review by jessicayuen — APPROVED** [review]
Thanks Jyoti!

File: internal/app/orchestrator/orchestrator.go (hunk 1 above)
Comment (jyotimahapatra) [reply internal/app/orchestrator/orchestrator.go:200]: There is no `x.Err` anymore since we just close the channel and do not send the error to caller.

Comment (jessicayuen) [reply internal/app/orchestrator/orchestrator.go:200]: What about the key? I think we should still include it in the log message

Comment (jyotimahapatra) [reply internal/app/orchestrator/orchestrator.go:200]: yeah,,key should be there..removing that was a mistake.

Comment (jyotimahapatra) [reply internal/app/orchestrator/orchestrator.go:200]: fixed it now

**Pull request merged**